### PR TITLE
add messagePayloadVersion

### PR DESCRIPTION
**Pull request by tbjump**

### CODE REVIEW

File: src/UniswapWormholeMessageReceiver.sol [hunk 1]
```diff
@@ -34,6 +34,7 @@ interface IWormhole {
 */
 contract UniswapWormholeMessageReceiver {
     string public constant NAME = "Uniswap Wormhole Message Receiver";
+    bytes32 constant expectedMessagePayloadVersion = keccak256(abi.encode("UniswapWormholeMessageSenderv1 (bytes32 receivedMessagePayloadVersion, address[] memory targets, uint256[] memory values, bytes[] memory datas, address messageReceiver, uint16 receiverChainId)"));
```
Comment (gator-boi) [inline src/UniswapWormholeMessageReceiver.sol:37]: Nit picky, but we should capitalize the V1 in `UniswapWormholeMessageSenderv1`.

File: src/UniswapWormholeMessageReceiver.sol [hunk 2]
```diff
@@ -97,7 +98,8 @@ contract UniswapWormholeMessageReceiver {
         require(vm.timestamp + MESSAGE_TIME_OUT_SECONDS >= block.timestamp, "Message no longer valid");
 
         // verify destination
-        (address[] memory targets, uint256[] memory values, bytes[] memory datas, address messageReceiver, uint16 receiverChainId) = abi.decode(vm.payload, (address[], uint256[], bytes[], address, uint16));
+        (bytes32 receivedMessagePayloadVersion, address[] memory targets, uint256[] memory values, bytes[] memory datas, address messageReceiver, uint16 receiverChainId) = abi.decode(vm.payload, (bytes32, address[], uint256[], bytes[], address, uint16));
+        require (expectedMessagePayloadVersion == receivedMessagePayloadVersion, "The payload version identifier of the Wormhole message does not match the expected one.");
```
Comment (gator-boi) [inline src/UniswapWormholeMessageReceiver.sol:102]: We should reduce the size of this revert string to less than 32 bytes.

File: src/UniswapWormholeMessageSender.sol [hunk 3]
```diff
@@ -19,13 +19,17 @@ interface IWormhole {
     function messageFee() external view returns (uint256);
 }
 
+bytes32 constant messagePayloadVersion = keccak256(abi.encode("UniswapWormholeMessageSenderv1 (bytes32 receivedMessagePayloadVersion, address[] memory targets, uint256[] memory values, bytes[] memory datas, address messageReceiver, uint16 receiverChainId)"));
+
+function generateMessagePayload(address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas, address _messageReceiver, uint16 _receiverChainId) pure returns(bytes memory) {
```
Comment (gator-boi) [inline src/UniswapWormholeMessageSender.sol:24]: This function should be declared as `internal`.

File: src/UniswapWormholeMessageReceiver.sol [hunk 4]
```diff
@@ -97,7 +98,8 @@ contract UniswapWormholeMessageReceiver {
         require(vm.timestamp + MESSAGE_TIME_OUT_SECONDS >= block.timestamp, "Message no longer valid");
 
         // verify destination
-        (address[] memory targets, uint256[] memory values, bytes[] memory datas, address messageReceiver, uint16 receiverChainId) = abi.decode(vm.payload, (address[], uint256[], bytes[], address, uint16));
+        (bytes32 receivedMessagePayloadVersion, address[] memory targets, uint256[] memory values, bytes[] memory datas, address messageReceiver, uint16 receiverChainId) = abi.decode(vm.payload, (bytes32, address[], uint256[], bytes[], address, uint16));
```
Comment (gator-boi) [inline src/UniswapWormholeMessageReceiver.sol:101]: Should this be changed to `calldatas` from `datas`?

File: src/UniswapWormholeMessageSender.sol (hunk 3 above)
Comment (tbjump) [reply src/UniswapWormholeMessageSender.sol:24]: It's a free function and as such doesn't have visibility. I made it a free function such that the test code can call it. 

I see a few alternatives, but not sure what the best practices in Solidity are: 
1. Make it part of the contract but then it either needs to be public/external(?) such that the test code can call it
2. the test needs to inherit UniswapWormholeMessageSender, which is a pattern I've seen in other places, but seems weird to me. 
3. Create another contract with only that method and have the Test as well as UniswapWormholeMessageSender inherit it. Seems like the cleanest method, but probably overkill.

Comment (tbjump) [reply src/UniswapWormholeMessageSender.sol:24]: I'll keep it as is for now.

**Pull request merged**